internal/guest/bridge/bridge_v2.go

```diff
@@ -133,18 +136,35 @@ func (b *Bridge) createContainerV2(r *Request) (_ RequestResponse, err error) {
 //
 // This is allowed only for protocol version 4+, schema version 2.1+
 func (b *Bridge) startContainerV2(r *Request) (_ RequestResponse, err error) {
-	_, span := trace.StartSpan(r.Context, "opengcs::bridge::startContainerV2")
+	ctx, span := trace.StartSpan(r.Context, "opengcs::bridge::startContainerV2")
 	defer span.End()
 	defer func() { oc.SetSpanStatus(span, err) }()
-	span.AddAttributes(trace.StringAttribute("cid", r.ContainerID))
+	span.AddAttributes(trace.StringAttribute(logfields.ContainerID, r.ContainerID))
 
 	// This is just a noop, but needs to be handled so that an error isn't
 	// returned to the HCS.
 	var request prot.MessageBase
-	if err := commonutils.UnmarshalJSONWithHresult(r.Message, &request); err != nil {
+	if err = commonutils.UnmarshalJSONWithHresult(r.Message, &request); err != nil {
 		return nil, errors.Wrapf(err, "failed to unmarshal JSON in message \"%s\"", r.Message)
 	}
 
+	// add container cgroup to OOM monitoring before starting the container
+	c, err := b.hostState.GetContainer(request.ContainerID)
+	if err == nil {
+		var cg cgroups.Cgroup
+		cg, err = c.GetCGroup(ctx)
+		if err != nil {
+			log.G(ctx).WithError(err).WithField(logfields.ContainerID, request.ContainerID).Error("could not get container cgroup")
+		} else {
+			if err = b.OomWatcher.Add(request.ContainerID, cg); err != nil {
+				log.G(ctx).WithError(err).WithField(logfields.ContainerID, request.ContainerID).Error("could not add container to OOM monitoring")
+			} else {
+				log.G(ctx).WithField(logfields.ContainerID, request.ContainerID).Debug("added container cgroup to OOM monitoring")
+			}
+		}
```

Comment:
We should probably just return the error here, if we can't load the cgroup that's not a good sign haha.

Reply:
do we also want to return the error if the cgroup cant be added to OOM monitoring?
or suppress that one, since its not critical and shouldnt fail starting a container?

Reply:
Adding to the OOM watcher failing should be fine to ignore

Reply:
"ignore" in the sense we still log it, but not returning the error seems fine. It's what they do in Containerd as well from looking.

Reply:
done